source/common/quic/quic_client_transport_socket_factory.h

```diff
@@ -43,8 +43,13 @@ class QuicClientTransportSocketFactory : public Network::CommonUpstreamTransport
   std::shared_ptr<quic::QuicCryptoClientConfig> getCryptoConfig() override;
 
 protected:
+  QuicClientTransportSocketFactory(
+      Ssl::ClientContextConfigPtr config,
+      Server::Configuration::TransportSocketFactoryContext& factory_context,
+      absl::Status& creation_status);
```

Comment:
Hm. I'm not in love with using an out-param in the constructor to indicate if the construction of this factory failed. What about having a bool member of `QuicClientTransportSocketFactory` which is something like `initialized` or `failed` or somesuch. Then the `create()` method can simply check that? (Or if we want the full Status, maybe have an `absl::Status` member named `status`?)

Reply:
yeah if I were designing this from scratch I'd go with doing the work initialize() but when I tried to do that I got chided for making member variables non-const.  The compromise folks came up with earlier was to make the constructors private (protected here for tests) and have a create method which explicitly fails so folks can't screw it up.

Reply:
Having a private constructor wrapped in a create method seems great. Totally sold on that. Weird that you got pushback on non-const members. i've never heard of that. Wacky. It wouldn't be too hard to make such a member const, if we wanted to.  But that being said, I'm not passionate about this so landing with this signature is fine with me.